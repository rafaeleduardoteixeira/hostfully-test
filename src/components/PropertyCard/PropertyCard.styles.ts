import styled from 'styled-components';
import { colors, media } from '../../themes';

export const PropertyCardContainer = styled.div`
  display: flex;
  flex-direction: column;
  gap: 10px;
  border: 1px solid #e0e0e0;
  border-radius: 10px;
  height: 310px;
  width: 300px;

  cursor: pointer;
  transition: box-shadow 0.2s cubic-bezier(0.2, 0, 0, 1);

  &:hover {
    box-shadow: 0 4px 8px rgba(0, 0, 0, 0.28);
  }

  @media ${media.tablet} {
    width: 400px;
  }

  @media ${media.mobile} {
    width: 330px;
  }
`;

export const PropertyCardImg = styled.img`
  border-radius: 10px 10px 0 0;
  height: 200px;
  width: 100%;
  border-radius: 10px;
  object-fit: cover;
`;

export const PropertyCardData = styled.div`
  display: flex;
  flex-direction: column;
  gap: 10px;
  padding: 10px 20px;
`;

export const PropertyLocation = styled.p`
  font-size: 1rem;
  font-weight: 600;
  color: ${colors.grayDark};
`;

export const PropertyName = styled.h3`
  font-size: 0.9rem;
  font-weight: 500;
  color: ${colors.gray};
`;

export const PropertyPrice = styled.p`
  font-size: 0.9rem;
  font-weight: 600;
  color: ${colors.grayDark};
`;
