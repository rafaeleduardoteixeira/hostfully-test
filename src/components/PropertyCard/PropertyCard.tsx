// Created a component called PropertyCard. It was created in a generic way to be used in different places.
import {
  PropertyCardContainer,
  PropertyCardData,
  PropertyCardImg,
  PropertyLocation,
  PropertyName,
  PropertyPrice,
} from './PropertyCard.styles';

interface PropertyCardProps {
  card: {
    id: string;
    images: string[];
    title: string;
    subTitle: string;
    description: string;
  };
  onClick: (id: string) => void;
}

export const PropertyCard = ({ onClick, card }: PropertyCardProps) => {
  return (
    <PropertyCardContainer key={card.id} onClick={() => onClick(card.id)}>
      <PropertyCardImg src={card.images[0]} alt={card.title} />
      <PropertyCardData>
        <PropertyLocation>{card.title}</PropertyLocation>
        <PropertyName>{card.subTitle}</PropertyName>
        <PropertyPrice>{card.description}</PropertyPrice>
      </PropertyCardData>
    </PropertyCardContainer>
  );
};
