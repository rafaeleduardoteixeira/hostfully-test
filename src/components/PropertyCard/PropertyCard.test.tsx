import React from 'react';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { PropertyCard } from './PropertyCard';

const mockedUsedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...(jest.requireActual('react-router-dom') as any),
  useNavigate: () => mockedUsedNavigate,
}));

describe('PropertyCard', () => {
  test('Snapshot test', () => {
    const { asFragment } = render(
      <PropertyCard
        onClick={jest.fn()}
        card={{
          id: '1',
          title: 'Title',
          images: ['image'],
          subTitle: 'SubTitle',
          description: 'Description',
        }}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
  test('Render', () => {
    render(
      <PropertyCard
        onClick={jest.fn()}
        card={{
          id: '1',
          title: 'Title',
          images: ['image'],
          subTitle: 'SubTitle',
          description: 'Description',
        }}
      />,
    );
    expect(screen.getByText('Title')).toBeInTheDocument();
    expect(screen.getByText('SubTitle')).toBeInTheDocument();
    expect(screen.getByText('Description')).toBeInTheDocument();
  });
  test('Click', () => {
    const onClick = jest.fn();
    render(
      <PropertyCard
        onClick={onClick}
        card={{
          id: '1',
          title: 'Title',
          images: ['image'],
          subTitle: 'SubTitle',
          description: 'Description',
        }}
      />,
    );
    screen.getByTestId('PropertyCard').click();
    expect(onClick).toHaveBeenCalled();
  });
});
