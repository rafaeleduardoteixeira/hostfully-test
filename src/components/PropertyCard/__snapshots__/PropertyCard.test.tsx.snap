// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyCard Snapshot test 1`] = `
<DocumentFragment>
  <div
    class="sc-braxZu hKguhO"
    data-testid="PropertyCard"
  >
    <img
      alt="Title"
      class="sc-gJhJTp gCpEdQ"
      src="image"
    />
    <div
      class="sc-elDIKY hNgXej"
    >
      <p
        class="sc-fQpRED clrPiF"
      >
        Title
      </p>
      <h3
        class="sc-dsLQwm cERdrA"
      >
        SubTitle
      </h3>
      <p
        class="sc-iKTcqh iNJjTT"
      >
        Description
      </p>
    </div>
  </div>
</DocumentFragment>
`;
